lightning/impl/sag.py

```diff
@@ -53,14 +77,42 @@ class SAGClassifier(BaseClassifier, _BaseSAG):
 
         minimize_w  1 / n_samples * \sum_i loss(w^T x_i, y_i)
                     + alpha * 0.5 * ||w||^2_2
+
+    Parameters
+    ----------
+    eta : float
+        step size for the gradient updates
+    alpha : float
+        amount of squared L2 regularization
+    beta : float
+        amount of regularization for the penalty term
+    loss : string
+        loss to use in the objective function. Can be one of
+        "smooth_hinge", "squared_hinge" or "log" (for logistic loss).
+    gamma : float
+        gamma parameter in the "smooth_hinge" loss (not used for other
+        loss functions)
+    max_iter : int
+        maximum number of outer iterations (also known as epochs).
+    tol : float
+        stopping criterion tolerance.
+    verbose : int
+        verbosity level. Set positive to print progress information.
+    callback : callable or None
+        if given, callback(self) will be called on each outer iteration
+        (epoch).
+    random_state: int or RandomState
+        Pseudo-random number generator state used for random sampling.
     """
 
-    def __init__(self, eta=1.0, alpha=1.0, loss="smooth_hinge", gamma=1.0,
-                 max_iter=10, n_inner=1.0, tol=1e-3, verbose=0,
-                 callback=None, random_state=None):
+    def __init__(self, eta=1.0, alpha=1.0, beta=0.0, loss="smooth_hinge",
+                 penalty=None, gamma=1.0, max_iter=10, n_inner=1.0, tol=1e-3,
```

Comment:
I think SAG should not have a keyword penalty, since you cannot set it to anything meaningful